test/e2e/e2e_suite_test.go

```diff
@@ -20,13 +22,27 @@ func TestE2E(t *testing.T) {
 }
 
 var _ = BeforeSuite(func() {
-	// FIXME: Since podman login doesn't work with daemonless image pulling, we need to login with docker first so podman tests don't fail.
-	if dockerUsername == "" || dockerPassword == "" {
-		// Test will be skipped anyway
+	if dockerUsername == "" && dockerPassword == "" && dockerHost == "" {
+		// No registry credentials or local registry host provided
+		// Skip tests
+		GinkgoT().Fatal("No registry credentials or local registry host provided")
+	}
+
+	if dockerHost != "" && dockerUsername == "" && dockerPassword == "" {
+		// running against local secure registry without credentials
+		// no need to login
 		return
 	}
 
-	dockerlogin := exec.Command("docker", "login", "-u", dockerUsername, "-p", dockerPassword, "quay.io")
-	err := dockerlogin.Run()
-	Expect(err).NotTo(HaveOccurred(), "Error logging into quay.io")
+	if dockerHost == "" {
+		// Default to Quay.io
+		dockerHost = "quay.io"
```

Comment:
nit: can we const this to `DEFAULT_REGISTRY` or something? It is easier to know the info on the top instead of burying somewhere in the middle. Not a big deal though.